R/create_shell_dataset.R

```diff
@@ -0,0 +1,138 @@
+#' @title Create Shell Dataset for Estimating Empirical Circumcision Rate
+#'
+#' @description  Create a shell dataset with a row for every unique area ID,
+#' area name, year and circumcision age in survey data. Also, computes the
+#' empirical number of person years until circumcision and number of people
+#' circumcised for several "types" of circumcision; known medical
+#' circumcisions, known traditional circumcisions, censored survey entries
+#' (i.e. where surveyed individuals had not been circumcised) and left-censored
+#'  survey entries (i.e. where circumcision occurred at an unknown age).
+#'
+#' @param survey_circumcision Information on male circumcision status from
+#' surveys.
+#' @param areas `sf` shapefiles for specific country/region.
+#' @param area_lev  PSNU area level for specific country. Defaults to the
+#' maximum area level found in `areas` if not supplied.
+#' @param time1 Variable name for time of birth, Default: "time1"
+#' @param time2 Variable name for time circumcised or censored,
+#' Default: "time2"
+#' @param strat Variable to stratify by in using a 3D hazard function,
+#' Default: "space"
+#' @param age - Variable with age circumcised or censored. Default: "age"
+#' @param circ Variables with circumcision matrix, Default: "indweight_st"
+#' @param ...  Further arguments passed to or from other methods.
+#'
+#' @seealso
+#'  \code{\link[threemc]{datapack_psnu_area_level}}
+#'  \code{\link[tidyr]{crossing}}
+#'  \code{\link[threemc]{create_integration_matrix_agetime}}
+#'  \code{\link[threemc]{create_hazard_matrix_agetime}}
+#' @return `data.frame` with a row for every unique record in
+#' `survey_circumcision` for a given area. Also includes empirical estimates
+#' for circumcision estimates for each unique record.
+#' @export
+#'
+#' @import dplyr
+#' @import sf
+#' @import rlang
```

Comment:
I would usually avoid importing all these packages and instead qualify the namespace when you call the functions. Avoids any chances of a clash between functions in packages and I think makes it easier to read when you come back to it in the future. Will have to import `%>%` explicitly though. Totally up to you though

Reply:
Thanks, good suggestion, amended with these changes. Have to import rlang though to use the rlang version of `.data`